worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -696,4 +722,68 @@ private void readBufferBySize(ByteBuffer buffer, int toRead) throws IOException
       }
     }
   }
+
+  public void cleanupExpiredShuffleKey(Set<String> expiredShuffleKeys) {
+    cleaner.cleanupExpiredShuffleKey(expiredShuffleKeys);
+  }
+}
+
+class PartitionFilesCleaner {
+  private static final Logger logger = LoggerFactory.getLogger(PartitionFilesCleaner.class);
+
+  private final LinkedBlockingQueue<PartitionFilesSorter.FileSorter> queue =
+      new LinkedBlockingQueue<>();
+  private final Lock lock = new ReentrantLock();
+  private final Condition notEmpty = lock.newCondition();
+
+  PartitionFilesCleaner() {
+    Thread cleaner =
+        new Thread(
+            () -> {
+              while (true) {
+                try {
+                  lock.lockInterruptibly();
+                  while (queue.isEmpty()) {
+                    notEmpty.await(500, TimeUnit.MILLISECONDS);
+                  }
+                  Iterator<PartitionFilesSorter.FileSorter> it = queue.iterator();
+                  while (it.hasNext()) {
+                    PartitionFilesSorter.FileSorter sorter = it.next();
+                    try {
+                      if (sorter.getOriginFileInfo().isStreamsEmpty()) {
```

Comment:
If there is no non skew streams, the origin file can be deleted. I think it's good to record all stream in fileinfo, I'll need it in supporting memory storage.

Reply:
> If there is no non skew streams, the origin file can be deleted.
it's not, only when the range open stream request(skew stream) will trigger `FileSorter.sort()`, and then the `FileSorter` instance will be added to the cleanup queue.

> I think it's good to record all stream in fileinfo, I'll need it in supporting memory storage.
yeah, I also read the proposal you posted on the mailing list, in the current implementation, record all activity streams in the fileinfo already done.

Reply:
the relations between `stream` and `fileinfo`:

for non-skew streams(non-range open stream request), it's `n` to `1`:

```
non-range stream id1
non-range stream id2  --->  original unsorted FileInfo A
non-range stream id3
```

for skew streams(range open stream request), it's `1` to `1`,  this is because we don't manage the sorted fileInfos within `StorageManager`:



```
range stream id4  --->  sorted FileInfo B
range stream id5  --->  sorted FileInfo C
range stream id6  --->  sorted FileInfo D
```